commons/src/main/kotlin/com/simplemobiletools/commons/helpers/BaseConfig.kt

```diff
@@ -549,4 +552,26 @@ open class BaseConfig(val context: Context) {
     var isCustomOrderSelected: Boolean
         get() = prefs.getBoolean(FAVORITES_CUSTOM_ORDER_SELECTED, false)
         set(selected) = prefs.edit().putBoolean(FAVORITES_CUSTOM_ORDER_SELECTED, selected).apply()
+
+    var viewType: Int
+        get() = prefs.getInt(VIEW_TYPE, VIEW_TYPE_LIST)
+        set(viewType) = prefs.edit().putInt(VIEW_TYPE, viewType).apply()
+
+    var contactsGridColumnCnt: Int
+        get() = prefs.getInt(
+            valueByConfiguration(CONTACTS_GRID_COLUMN_COUNT_PORTRAIT, CONTACTS_GRID_COLUMN_COUNT_LANDSCAPE),
+            valueByConfiguration(
+                context.resources.getInteger(R.integer.contacts_grid_columns_count_portrait),
+                context.resources.getInteger(R.integer.contacts_grid_columns_count_landscape)
+            )
+        )
+        set(contactsGridColumnCnt) = prefs.edit().putInt(
+            valueByConfiguration(CONTACTS_GRID_COLUMN_COUNT_PORTRAIT, CONTACTS_GRID_COLUMN_COUNT_LANDSCAPE),
+            contactsGridColumnCnt
+        ).apply()
+
+    private fun <T> valueByConfiguration(portraitValue: T, landScapeValue: T): T {
```

Comment:
this is a really complicated solution to an easy problem.. Just check how is portraitColumnCnt and landscapeColumnCnt done in App Launcher, that is a lot easier to read.

Reply:
I have simplified the function. Could you please take a look at the implementation now?

Reply:
this should be good enough, renamed some things a bit